modules/analysis/src/service.rs

```diff
@@ -371,7 +390,23 @@ impl AnalysisService {
                         .node_indices()
                         .filter(|&i| {
                             if let Some(node) = graph.node_weight(i) {
-                                node.name.contains(&query.q.to_string())
+                                if let Some(re_name) = graph_query_map.get("re_name") {
+                                    // if no specific url params supplied then use contains search
+                                    node.name.contains(re_name)
+                                } else {
+                                    // if any specific url params supplied then match equals
+                                    let matches_sbom_id = graph_query_map
+                                        .get("sbom_id")
+                                        .map_or(true, |sbom_id| node.sbom_id.eq(sbom_id));
+                                    let matches_node_id = graph_query_map
+                                        .get("node_id")
+                                        .map_or(true, |node_id| node.node_id.eq(node_id));
+                                    let matches_name =
+                                        graph_query_map.get("name").map_or(true, |name| {
+                                            !name.is_empty() && node.name.eq(name)
+                                        });
+                                    matches_sbom_id && matches_node_id && matches_name
+                                }
```

Comment:
This lump of code is repeated below, which is crying out for a refactor into a single, common fn. If it were me, I'd probably rename/modify `convert_query_to_hashmap` to take both a `Query` and a `Node` and then return a `bool`, and call it here. 

I'd prolly use regex's instead of the temporary hashmap, too, so feel free to ignore me. :smile:

Reply:
this area may evolve depending on @dejanb usage ... so will leave it for now.